pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,826 @@
+    filename : str
+        Path to the filename to set the lens sets list to
+    make_backup : bool
+        To indicate if a backup file should be created or not. Default = `True`
+    """
+    # Make a backup with today's date.
+    if make_backup:
+        backup_path = filename + str(date.today()) + '.bak'
+        try:
+            shutil.copyfile(filename, backup_path)
+        except Exception as ex:
+            logger.error('Something went wrong with copying the file %s', ex)
+            pass
+    with open(filename, 'wb') as lens_file:
+        np.save(lens_file, np.array(sets_list_of_tuples, dtype=object),
+                allow_pickle=True)
+
+
+def get_att_len(energy, material="Be", density=None):
+    """
+    Get the attenuation length (in meter) of a material.
+
+    The X-ray beam intensity I(x) at depth x in a material is a function of the
+    attenuation coefficient mu, and can be calculated by the Beer-Lambert law.
+    The Absorption Length (or Attenuation Length) is defined as the distance
+    into a material where the x-ray beam intensity has decreased to a value of
+    1/e (~ 40%) of the incident beam intensity (Io).
+
+    (1/e) = e^(-mu * x)
+    ln(1/e) = ln(e^(-mu * x))
+    1 = mu * x
+    x = 1/mu
+
+    Usage:
+
+    .. code-block:: python
+
+        get_att_len(energy=8, material='Be')
+
+    Parameters
+    ----------
+    energy : number
+        Beam Energy given in KeV
+    material : str
+        Default - Beryllium.  The use of beryllium extends the range of
+        operation of compound refractive lenses, improving transmission,
+        aperture size, and gain.
+    density : float
+        Material density in g/cm^3
+
+    Returns
+    -------
+    att_len : float
+        Attenuation length
+    """
+    try:
+        # xdb.material_my returns absorption length in 1/cm and takes energy
+        # or array of energies in eV.
+        att_len = 1.0 / (xdb.material_mu(material, energy * 1.0e3,
+                                         density=density)) * 1.0e-2
+
+    except Exception as ex:
+        logger.error('Get Attenuation Length error: %s', ex)
+        return
+    # TODO: if we return we'll get att_len == None, is this what we want?
+    return att_len
+
+
+# TODO: do we still need the density here? we are now
+# using density=xdb.atomic_density(material)? or should I
+# use xdb.atomic_density only if density == None?
+def get_delta(energy, material="Be", density=None):
+    """
+    Calculate delta for a given material at a given energy.
+
+    Anomalous components of the index of refraction for a material, using the
+    tabulated scattering components from Chantler.
+
+    Usage:
+
+    .. code-block:: python
+
+        get_delta(energy=8, material='Au')
+
+    Parameters
+    ----------
+    energy : number
+        x-ray energy in KeV
+    material : str
+        Chemical formula (‘Fe2O3’, ‘CaMg(CO3)2’, ‘La1.9Sr0.1CuO4’)
+        Default - Beryllium.
```

Comment:
Does only take chemical formulas? This would be useful to mention in the other functions instead of or in addition to the note about its benefits. This would also mean that 'Beryllium' could be misleading.
```suggestion
    material : str
        Chemical formula (‘Fe2O3’, ‘CaMg(CO3)2’, ‘La1.9Sr0.1CuO4’)
        Defaults to 'Be'.
```
If not, doesn't matter as much but it would still be nice to know what formats it accepts.

Reply:
I also noticed your quotes here are actual quotes `‘` instead of apostrophes `'`. I'm guessing these are straight from https://xraypy.github.io/XrayDB/python.html#xraydb.xray_delta_beta and it should be fine in this context since it's in the docstring but I just thought it was interesting 🤔

Reply:
So I am actually wrong here, thanks for pointing this out - it does not take a chemical formula, it takes an atomic symbol for element - i'll correct this. And thanks for the suggestion about `Be`

Reply:
fixed it in 0f33d1e